### ISSUE TITLE
Security: fail closed when TTY raw mode cannot be verified

### ISSUE THREAD

**Issue by tim-watcha** [dazebug/terminal-checkout#22]
## Summary

Claude input delivery currently treats an unavailable or unrecognized `stty` result as if the TTY were in raw mode.

In `app/Sources/Core/ClaudeInjector.swift`, `acceptingClaudePID` contains:

```swift
guard ttyIsRawMode(sttyOutput: sttyOutput) ?? true else { return nil }
```

`probeAcceptingClaudePID` also converts an `stty` failure into an empty string, which reaches the same fail-open path. The current test `testAcceptsInputFallsBackToForegroundWhenSttyUnavailable` explicitly preserves this behavior.

This conflicts with the documented three-part safety gate for Claude injection:

1. The foreground process is Claude.
2. The TTY is in raw mode.
3. The PID still matches the original Claude session.

## Impact

If `stty` fails or its output format becomes unrecognized, the app proceeds using only the foreground-process check. During the canonical startup interval, kernel echo can make typed text appear on screen and cause the reflection check to produce a false positive. This can result in premature input or carriage-return delivery, lost input, or residual input reaching a later reader during a session transition.

## Proposed change

Require an affirmative raw-mode result:

```swift
guard ttyIsRawMode(sttyOutput: sttyOutput) == true else { return nil }
```

Treat process failure, nonzero status, empty output, and unrecognized output as not ready, then continue polling until the timeout.

## Acceptance criteria

- `acceptingClaudePID` returns `nil` when raw mode is unknown.
- `probeAcceptingClaudePID` does not convert an `stty` failure into an accepted state.
- Tests cover empty output, command failure, unrecognized output, canonical mode, and raw mode.
- No bytes are sent unless all three documented gates pass.
